Client/TeamTalkAndroid/src/main/java/dk/bearware/backend/TeamTalkService.java

```diff
@@ -622,8 +622,7 @@ private void login() {
 
         String nickname = ttserver.nickname;
         if (TextUtils.isEmpty(nickname)) {
-            String def_nick = getResources().getString(R.string.pref_default_nickname);
-            nickname = PreferenceManager.getDefaultSharedPreferences(getApplicationContext()).getString(Preferences.PREF_GENERAL_NICKNAME, def_nick);
+            nickname = PreferenceManager.getDefaultSharedPreferences(getApplicationContext()).getString(Preferences.PREF_GENERAL_NICKNAME);
```

Comment:
You should change R.string.pref_default_nickname to blank instead of removing the code that allows a default value.